### ISSUE TITLE
verify that phone numbers from source are being processed and transformed correctly

### ISSUE THREAD

**Issue by eric-jahn** [211tbc/synthesis#6]
We received a report that Suncoast Center was not ever receiving client phone numbers in the TBC extended XML, but this may be because the source system is not including them.

**Comment by yd-adept** [comment]
I looked into this using the test file HUD_HMIS_TBC.xml. This file is missing the xml tag SourceContactPhone so I added it. When synthesis shreds a document which contains this tag, the field is stored in the database. However the source phone number is not part of the data submitted to NetSmart. Is that by design?

**Comment by eric-jahn** [comment]
No I think it's unintended, and it should be sent to Netsmart.

**Comment by eric-jahn** [comment]
Oh, wait.  SourceContactPhone is something entirely different.  It's the phone number of the agency contact who is transmitting that XML file, and Netsmart should not receive it.  What Netsmart should receive is the PersonPhoneNumber, which they can call back with: https://github.com/211tbc/synthesis/blob/64817b83fd67664f579fd32ed976d3857b7b7362/src/test_files/HUD_HMIS_TBC.xml#L74
Bowman may also start populating "Caller Home Phone" https://github.com/211tbc/synthesis/blob/master/src/test_files/HUD_HMIS_TBC.xml#L171 which is the same thing for all intents and purposes.  Both go in the same place in the transformed doc.  I'm asking Bowman to just use the new Caller Home Phone field.

**Comment by yd-adept** [comment]
OK. The PersonPhoneNumber is detected and stored in the database. In the document transmitted to Netsmart, the xml tag telecom contains the phone number.

If "Caller Home Phone" will be used in the transformed document then we willl need to change the hl7CCDwriter module to make use of it now.

**Comment by eric-jahn** [comment]
Yes, exactly.  Had the source system generally been sending us the PersonPhoneNumber?  As I understand it, it was sporadic, to the extent Suncoast Center asserted they never received the numbers.

**Comment by yd-adept** [comment]
In tbc-pryamid.old, I ran a grep command searching for "PersonPhoneNumber" in the used_files folder. 116 out of 163 files contain the text.

Right now I'm looking at the synthesis database on pix.penguix.net.

The database table "person_historical" is the table that contains the "person_phone_number" field. When I perform a select of that table and page through the results, the only fields I see populated are the "id", "person_index_id" and "site_service_participation_index_id".  All other fields are empty! There is definitely something wrong.

I did preserve the old tbc-pyramid install as "tbc-pyramid.old" and that virtualenv may still work even with server updates. If you think it is worth while, I can debug the virtualenv on the server to figure out what is happening otherwise I'll work on changing the hl7CCDwriter module.

**Comment by eric-jahn** [comment]
Yes, please debug the virtualenv to figure out what is going on .  This
could be our source of problems on this issue.

**Comment by yd-adept** [comment]
After being stumped for a couple of days, it finally dawned on me that I was looking at the wrong database (synthesis instead of tbcsynthesis). In tbcsynthesis, the person_historical table does contain the "PersonPhoneNumber" values from the xml documents shredded that I tested and the phone numbers are present in the XML attachments sent to Netsmart,

I have to believe now that if there are complaints about the phone number being missing then it has to be missing from the incoming XML.

I'll start on the changes to the hl7CCDwriter module soon.
